### PR TITLE
Bugfix DC-296 - Socure data validation/IAL elevation

### PR DESCRIPTION
**Pull request by necampanini**
# PR Review Priority Guide — DC-296

- **Branch:** `bugfix/DC-296-socure-data-validation` -> `main`
- **State:** pre-PR (no #)
- **Risk Level:** Medium-High
- **Files:** 30 | **Lines:** +2,207 / -127
- **Generated:** 2026-04-24

Catches Socure outbound payload shape errors (10-digit SSN/ITIN, non-E.164 phone, oversized names, reused evaluation ids) before they 400 the workflow, plus routes DocV webhook outcomes off the top-level workflow decision and refreshes JWT claims from DB post-DocV. Reviewer attention belongs on the claim-refresh logic, the webhook decision-routing shift, and the boundary validation guards.

---

#### 🔗 Jira ticket
- [DC-296](https://codeforamerica.atlassian.net/browse/DC-296)

#### ✍️ Description
Hardens the data validation on the outbound path to Socure so shape errors never reach the workflow. Frontend Zod schema enforces 9-digit SSN/ITIN and calendar-valid DOB. Backend adds the same SSN/ITIN guard in `SubmitIdProofingCommandHandler`, normalizes phones to E.164 and truncates names to 240 chars at the `HttpSocureClient` boundary, and sends a unique `id` per evaluation. DocV webhook processing routes on the top-level workflow decision (ACCEPT/REJECT/RESUBMIT/REVIEW) rather than the document-enrichment decision. Session JWT refresh re-pulls `ial` and `id_proofing_status` from the DB when the DB reflects a post-DocV elevation, using a rank comparison to block any accidental downgrade. Also cleans up offboarding copy for decline-consent and normalizes `SocureExcludedIdentifierTypes`.

#### 🔗 Links to related PRs
- Connector changes: none (portal-only)
- Follow-ups tracked: DC-138 resubmit handling (deferred), offboarding copy distinctness (tracked separately), Medicaid ID shape (open product question)

#### ✅ Completion tasks
- [x] Added relevant tests
- [x] Meets acceptance criteria in ticket
- [ ] Configuration changes:
  - [ ] If new environment variables are added, update in Tofu
  - [ ] If new environment secrets are added, update in Tofu and set in AWS Secret Manager
  - [ ] If you're adding an appsetting, add it to the requisite example file, and update it in the AWS AppConfig
  - [ ] If appsetttings are changed, update in AWS AppConfig

No config changes in this branch. One NuGet dep added: `libphonenumber-csharp` (matches CO connector version).

---

## Priority Review Table

| Priority | File | Change | Lines | Sec | Cpx | Nov | Notes |
|:---:|---|---|:---:|:---:|:---:|:---:|---|
| 🔴 | `src/SEBT.Portal.Infrastructure/Services/JwtTokenService.cs` | `GenerateForSessionRefresh` IAL/status elevation via `RankIal` | 205-279 | High | Med | Med | Claim downgrade defense |
| 🔴 | `src/SEBT.Portal.UseCases/IdProofing/ProcessWebhook/ProcessWebhookCommandHandler.cs` | Switch to workflow decision; RESUBMIT/REVIEW -> Rejected | 80-220 | High | Med | High | Routing behavior change |
| 🔴 | `src/SEBT.Portal.UseCases/IdProofing/SubmitIdProofing/SubmitIdProofingCommandHandler.cs` | SSN/ITIN 9-digit guard before Socure call | 58-72, 388-416 | High | Low | Low | Data boundary check |
| 🔴 | `src/SEBT.Portal.Infrastructure/Services/HttpSocureClient.cs` | E.164 phone + name truncate + unique evaluation id | 45-68, 193-225 | High | Med | Med | Per-eval id is a semantic shift |
| 🔴 | `src/SEBT.Portal.Web/src/features/auth/api/submit-id-proofing/schema.ts` (new) | DOB calendar/age/future + SSN/ITIN digit validation | 1-90 | High | Med | Med | PII-adjacent input contract |
| 🟡 | `src/SEBT.Portal.Web/src/features/auth/components/id-proofing/IdProofingForm.tsx` | Per-option `validation`, digit-stripping, shape errors | 11-180 | High | Med | Med | Untranslated copy (TODOs) |
| 🟡 | `src/SEBT.Portal.Infrastructure/Services/PhoneNormalizer.cs` (new) | libphonenumber US parse/validate wrapper | 1-36 | Low | Low | Med | Mirrors CO connector API |
| 🟡 | `src/SEBT.Portal.Infrastructure/Socure/SocureExcludedIdentifierTypes.cs` | Excluded-type filter adjustments | full | Med | Low | Low | Affects which ids reach Socure |
| 🟡 | `src/SEBT.Portal.Web/src/features/auth/components/doc-verify/DocVerifyPage.tsx` | Post-verify state handling | full | Med | Med | Low | Pairs with JWT refresh |
| 🟡 | `src/SEBT.Portal.Web/src/features/auth/components/off-boarding/OffBoardingPage.tsx` | Decline-consent reason/copy plumbing | full | Low | Low | Med | Copy gap still open |
| 🟢 | `*.test.*` (8 files, ~1,300 lines) | Coverage for all above | — | Low | — | — | Skim for gap coverage |
| 🟢 | `WebhookController.cs`, `ProcessWebhookCommand.cs` | Pass-through plumbing for `workflowDecision` | — | Low | Low | Low | Wire-up |
| 🟢 | `mocks/handlers.ts`, `HouseholdFactory.cs`, `useSubmitIdProofing.test.tsx` | Fixture tweaks | — | Low | Low | Low | Trivial |
| 🟢 | `SEBT.Portal.Infrastructure.csproj` | `libphonenumber-csharp` dep | +1 | Low | — | Low | Matches CO version |

---

## Review Recommendation

Start with the four red items in this order:

1. **`JwtTokenService.cs`** — verify `RankIal` ordering (`2 > 1plus > 1 > null`) is correct, and that `IdProofingCompletedAt` / `IdProofingExpiresAt` claims are always repopulated whenever `IdProofingStatus` is set to `Completed` (`BuildAndSignToken` invariant depends on it). Confirm the non-Completed branch can never silently drop an existing higher-value status claim.
2. **`ProcessWebhookCommandHandler.cs`** — the switch from `documentDecision` to `workflowDecision` changes observable behavior: RESUBMIT and REVIEW now terminate as Rejected rather than staying Pending. Verify every code path that previously relied on "review" / "resubmit" being intermediate is either gone or intentionally re-routed; confirm `evaluation_paused` still the only intermediate path.
3. **`SubmitIdProofingCommandHandler.cs`** (guard) + **`HttpSocureClient.cs`** (boundary) — read as a pair. The 9-digit check is the gating fence, the E.164 / truncate / unique-id transforms are what ships when the fence passes. Confirm `Guid.NewGuid()` on the top-level `id` doesn't break any downstream correlation we rely on.
4. **`schema.ts`** — the DOB `isRealCalendarDate` trick (JS Date rollover) is correct but non-obvious; cross-check against `schema.test.ts` leap-year and month-boundary cases.

Yellow items: `IdProofingForm.tsx` has three user-facing strings hardcoded with `// TODO: Use t(...)` — confirm DC CSV gap is tracked. `SocureExcludedIdentifierTypes` change is small but affects which id types reach the API; verify it matches the intended include/exclude set.

Tests (~1,300 lines) can be skimmed for coverage rather than read line-by-line. The ones worth opening: `HttpSocureClientTests.cs` (+297 — name truncation + phone edge cases), `ProcessWebhookCommandHandlerTests.cs` (+221 — RESUBMIT/REVIEW now-terminal), `SessionRefreshTokenServiceTests.cs` (+124 — IAL rank rules), `schema.test.ts` (+197, new — DOB edge cases).

[DC-296]: https://codeforamerica.atlassian.net/browse/DC-296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/SEBT.Portal.Infrastructure/Services/Socure/SocureExcludedIdentifierTypes.cs [hunk 1]
```diff
@@ -4,12 +4,18 @@ namespace SEBT.Portal.Infrastructure.Services.Socure;
 
 /// <summary>
 /// Benefit-program identifiers are not sent on Socure evaluation requests; see <see cref="IdProofingBenefitIdentifierTypes"/>.
+/// Medicaid IDs are also excluded because they do not match Socure's <c>national_id</c> schema
```
Comment (noxmwalsh) [inline src/SEBT.Portal.Infrastructure/Services/Socure/SocureExcludedIdentifierTypes.cs:7]: Yeah, we should avoid sending them that.  I don't think Medicaid is supported as part of Socure's validation (or at the least not with our current contract with them)

File: src/SEBT.Portal.Infrastructure/Services/Socure/SocureExcludedIdentifierTypes.cs [hunk 2]
```diff
@@ -4,12 +4,18 @@ namespace SEBT.Portal.Infrastructure.Services.Socure;
...
+/// (exactly 4 or 9 digits), so they stay as a state-plugin-only identifier.
 /// </summary>
 internal static class SocureExcludedIdentifierTypes
 {
     /// <summary>
-    /// When true, <c>national_id</c> must not be set for this portal id type (SNAP/TANF stay in-portal).
+    /// When true, <c>national_id</c> must not be set for this portal id type.
+    /// SNAP/TANF stay in-portal (matched against co-loaded records) and Medicaid is excluded
```
Comment (noxmwalsh) [inline src/SEBT.Portal.Infrastructure/Services/Socure/SocureExcludedIdentifierTypes.cs:14]: This is also accurate; we only match with DC's guardian records for SNAP/TANF.

File: src/SEBT.Portal.Infrastructure/Services/PhoneNormalizer.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+using PhoneNumbers;
+
+namespace SEBT.Portal.Infrastructure.Services;
+
+/// <summary>
+/// Normalizes phone numbers to 10-digit US national format using libphonenumber.
+/// Ported from the CO connector's PhoneNormalizer to keep parsing behavior consistent across plugins.
+/// </summary>
+internal static class PhoneNormalizer
```
Comment (noxmwalsh) [inline src/SEBT.Portal.Infrastructure/Services/PhoneNormalizer.cs:9]: 👍🏻

File: src/SEBT.Portal.UseCases/IdProofing/ProcessWebhook/ProcessWebhookCommandHandler.cs [hunk 4]
```diff
@@ -189,21 +195,23 @@ private bool ValidateWebhookSignature(string? bearerToken)
...
-
-    private static DocVerificationStatus? MapDecisionToStatus(string? decision)
+    private static DocVerificationStatus? MapWorkflowDecisionToStatus(string? decision)
     {
-        return decision?.ToLowerInvariant() switch
+        return decision?.ToUpperInvariant() switch
         {
-            "accept" => DocVerificationStatus.Verified,
-            "reject" => DocVerificationStatus.Rejected,
+            "ACCEPT" => DocVerificationStatus.Verified,
```
Comment (noxmwalsh) [inline src/SEBT.Portal.UseCases/IdProofing/ProcessWebhook/ProcessWebhookCommandHandler.cs:211]: Oh yikes, was this meant be to uppercase?  Good catch!

File: src/SEBT.Portal.Web/src/app/(public)/login/id-proofing/off-boarding/page.tsx [hunk 5]
```diff
@@ -19,13 +19,34 @@ export default async function OffBoardingPage({ searchParams }: OffBoardingPageP
...
+
   return (
     <div className="usa-section">
       <div className="grid-container maxw-tablet">
         <section aria-labelledby="off-boarding-title">
           <OffBoardingContent
-            title={t('title')}
-            body={t('body1')}
+            title={title}
+            body={body}
```
Comment (noxmwalsh) [inline src/SEBT.Portal.Web/src/app/(public)/login/id-proofing/off-boarding/page.tsx:49]: Are we OK with losing localization for this?

**Review by noxmwalsh — APPROVED** [review]
Approved with non-blocking comments
